src/prommis/uky/costing/custom_costing_example.py

```diff
@@ -0,0 +1,171 @@
+
+        material_factor_dict = (
+            {  # material factors for each valid choice of shell material
+                "carbonsteel": 1,
+                "stainlessteel": 1.5,
+                "aluminum": 2,
+            }
+        )
+
+        blk.capital_cost_per_unit = Var(
+            initialize=1000,
+            units=CE_index_units,
+            domain=NonNegativeReals,
+            bounds=(0, None),
+        )
+
+        @blk.Constraint()
+        def capital_cost_per_unit_eq(blk):
+            # cost equation CAPITAL_COST = REF_COST * (VOLUME / REF_VOLUME)**0.6
+            ref_cost = 10000 * pyunits.USD_custom
+            ref_volume = 1000 * pyunits.m**3
+            return blk.capital_cost_per_unit == pyunits.convert(
+                material_factor_dict[material]
+                * ref_cost
+                * (volume_per_unit / ref_volume) ** 0.6,
+                to_units=CE_index_units,
+            )
+
+        # create a variable capital_cost that the REE Costing Framework can look for
+        blk.capital_cost = Var(
+            initialize=1000,
+            units=CE_index_units,
+            domain=NonNegativeReals,
+            bounds=(0, None),
+        )
+
+        @blk.Constraint()
+        def capital_cost_constraint(blk):
+            return blk.capital_cost == blk.capital_cost_per_unit * blk.number_of_units
+
+        # define the fixed costs
+
+        blk.fixed_operating_cost_per_unit = Var(
+            initialize=1000,
+            units=CE_index_units,
+            domain=NonNegativeReals,
+            bounds=(0, None),
+        )
+
+        @blk.Constraint()
+        def fixed_operating_cost_per_unit_eq(blk):
+            return blk.fixed_operating_cost_per_unit == pyunits.convert(
+                0.05 * blk.capital_cost, to_units=CE_index_units
+            )
+
+        # create a variable fixed_operating_cost that the REE Costing Framework can look for
+        blk.fixed_operating_cost = Var(
+            initialize=1000,
+            units=CE_index_units,
+            domain=NonNegativeReals,
+            bounds=(0, None),
+        )
+
+        @blk.Constraint()
+        def fixed_operating_cost_constraint(blk):
+            return (
+                blk.fixed_operating_cost
+                == blk.fixed_operating_cost_per_unit * blk.number_of_units
+            )
+
+        # define the variable costs
+
+        blk.variable_operating_cost_per_unit = Var(
+            initialize=1000,
+            units=CE_index_units / pyunits.year,
+            domain=NonNegativeReals,
+            bounds=(0, None),
+        )
+
+        @blk.Constraint()
+        def variable_operating_cost_per_unit_eq(blk):
+            return blk.variable_operating_cost_per_unit == pyunits.convert(
+                0.00190
+                * pyunits.USD_custom
+                / pyunits.gal
+                * water_injection_rate_per_unit,
+                to_units=CE_index_units / pyunits.year,
```

Comment:
Rather than `pyunits.year`, should this be the the global parameter defined for `base_year`?

Reply:
@mollydougher I took a look at your membrane costing model, and I see that it references the costing package as `blk.costing_package.base_currency` and `blk.costing_package.base_period`. This is nice because it will always reference the flowsheet block whether `QGESSCosting()` or a custom class is used. I will update this example to use that convention, thank you for the suggestion!